pysb/simulator/bng.py

```diff
@@ -74,7 +74,8 @@ def run(self, tspan=None, initials=None, param_values=None, n_runs=1,
         """
         super(BngSimulator, self).run(tspan=tspan,
                                       initials=initials,
-                                      param_values=param_values
+                                      param_values=param_values,
+                                      _run_kwargs=locals()
```

Comment:
This seems hacky. At the very least, `Simulator.run` needs to document the contract that subclasses need to follow when overriding it.

An alternative would be to make all `run` methods take and propagate `**kwargs`, and write a helper to manage setting default values, but of course the `run` signatures will be less useful without the explicit arg names and defaults.

Reply:
I've updated the `Simulator.run()` documentation to reflect the suggested contract with the `_run_kwargs` argument. If it's not supplied, a warning is now generated.